src/ray/raylet/scheduling_queue.cc

```diff
@@ -223,6 +223,32 @@ void SchedulingQueue::QueueMethodsWaitingForActorCreation(
   QueueTasks(methods_waiting_for_actor_creation_, tasks);
 }
 
+bool SchedulingQueue::TaskInQueue(const TaskID &task_id) const {
+  if (methods_waiting_for_actor_creation_.TaskInQueue(task_id)) {
```

Comment:
with the underlying map implementation the performance effect will be minor, but I would check the queues in the order that checks the longest queues last (e.g., running, ready, waiting, placeable).